File: backend/spring-boot-kotlin/src/main/java/org/bootstrapbugz/api/admin/payload/request/UserRequest.java
package org.bootstrapbugz.api.admin.payload.request;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import java.util.Set;
import lombok.Builder;
import org.bootstrapbugz.api.shared.constants.Regex;
import org.bootstrapbugz.api.shared.validator.FieldMatch;
import org.bootstrapbugz.api.user.model.Role.RoleName;

@Builder
@FieldMatch(first = "password", second = "confirmPassword", message = "{passwords.doNotMatch}")
public record UserRequest(
    @NotBlank(message = "{firstName.required}")
        @Pattern(regexp = Regex.FIRST_AND_LAST_NAME, message = "{firstName.invalid}")
        String firstName,
    @NotBlank(message = "{lastName.required}")
        @Pattern(regexp = Regex.FIRST_AND_LAST_NAME, message = "{lastName.invalid}")
        String lastName,
    @NotBlank(message = "{username.required}")
        @Pattern(regexp = Regex.USERNAME, message = "{username.invalid}")
        String username,
    @NotBlank(message = "{email.required}")
        @Email(message = "{email.invalid}", regexp = Regex.EMAIL)
        String email,
    @NotBlank(message = "{password.required}")
        @Pattern(regexp = Regex.PASSWORD, message = "{password.invalid}")
        String password,
    @NotBlank(message = "{password.required}")
        @Pattern(regexp = Regex.PASSWORD, message = "{password.invalid}")
        String confirmPassword,
    @NotNull(message = "{user.active.required}") Boolean active,
    @NotNull(message = "{user.lock.required}") Boolean lock,
    @NotEmpty(message = "{roles.empty}") Set<RoleName> roleNames) {}

File: backend/spring-boot-kotlin/src/main/java/org/bootstrapbugz/api/admin/payload/request/UserRequest.kt
package org.bootstrapbugz.api.admin.payload.request

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotEmpty
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Pattern
import org.bootstrapbugz.api.shared.constants.Regex
import org.bootstrapbugz.api.shared.validator.FieldMatch
import org.bootstrapbugz.api.user.model.Role.RoleName

@FieldMatch(first = "password", second = "confirmPassword", message = "{passwords.doNotMatch}")
data class UserRequest(
    @NotBlank(message = "{firstName.required}")
    @field:Pattern(regexp = Regex.FIRST_AND_LAST_NAME, message = "{firstName.invalid}")
    val firstName: String,
    @NotBlank(message = "{lastName.required}")
    @field:Pattern(regexp = Regex.FIRST_AND_LAST_NAME, message = "{lastName.invalid}")
    val lastName: String,
    @NotBlank(message = "{username.required}")
    @field:Pattern(regexp = Regex.USERNAME, message = "{username.invalid}")
    val username: String,
    @NotBlank(message = "{email.required}")
    @field:Email(message = "{email.invalid}", regexp = Regex.EMAIL)
    val email: String,
    @NotBlank(message = "{password.required}")
    @field:Pattern(regexp = Regex.PASSWORD, message = "{password.invalid}")
    val password: String,
    @NotBlank(message = "{password.required}")
    @field:Pattern(regexp = Regex.PASSWORD, message = "{password.invalid}")
    val confirmPassword: String,
    @NotNull(message = "{user.active.required}")
    val active: Boolean,
    @NotNull(message = "{user.lock.required}")
    val lock: Boolean,
    @NotEmpty(message = "{roles.empty}")
    val roleNames: MutableSet<RoleName>
)
